src/context/partition_unit.rs

```diff
@@ -288,7 +290,8 @@ impl<'a> ContextWriter<'a> {
     } else {
       r = if prev_ul == prev_u { prev_u } else { prev_l };
     }
-    (r as u8, cdf_index)
+
+    ((r as u8).min(last_active_segid), cdf_index)
```

Comment:
Was this a bug in our context code?

Reply:
Kind of yes, kind of no. It previously never triggered any odd behavior due to the fixed number of segments/probably some amount of luck. With the dynamic segment selection, this was causing issues with skip blocks (see line 442).